core/discovery/brokerdiscovery/repository.go

```diff
@@ -86,18 +85,14 @@ func (s *Repository) Start() error {
 		return err
 	}
 
-	s.stopOnce = sync.Once{}
-	s.stopChan = make(chan bool)
 	go s.proposalWatchdog()
 
 	return nil
 }
 
 // Stop ends proposals synchronization to storage
 func (s *Repository) Stop() {
-	s.stopOnce.Do(func() {
-		close(s.stopChan)
-	})
+	s.stopChan <- struct{}{}
```

Comment:
why? this means only a single listener will receive the signal, do close it.

Reply:
There is just one listener. I dont get what will it give, because then you have to do:
```
pr.once.Do(func() {
		log.Info().Msg("Stopping provider registrar")
		close(pr.stopChan)
	})
```

It was added to handle multiple stops.

Reply:
I don't understand what you've written.

Reply:
Rephrased

Reply:
Added clearing opened channels